src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/DossierContextFactory.kt

```diff
@@ -0,0 +1,70 @@
+package uk.gov.justice.digital.hmpps.managerecallsapi.service
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.managerecallsapi.controller.PrisonLookupService
+import uk.gov.justice.digital.hmpps.managerecallsapi.controller.SearchRequest
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.Recall
+import uk.gov.justice.digital.hmpps.managerecallsapi.db.RecallRepository
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.FirstName
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.LastName
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.MiddleNames
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.NomsNumber
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.PrisonName
+import uk.gov.justice.digital.hmpps.managerecallsapi.domain.RecallId
+import uk.gov.justice.digital.hmpps.managerecallsapi.search.Prisoner
+import uk.gov.justice.digital.hmpps.managerecallsapi.search.PrisonerOffenderSearchClient
+
+@Component
+class DossierContextFactory(
+  @Autowired private val recallRepository: RecallRepository,
+  @Autowired private val prisonLookupService: PrisonLookupService,
+  @Autowired private val prisonerOffenderSearchClient: PrisonerOffenderSearchClient
+) {
+  fun createContext(recallId: RecallId): DossierContext {
+    val recall = recallRepository.getByRecallId(recallId)
+    val prisoner = prisonerOffenderSearchClient.prisonerSearch(SearchRequest(recall.nomsNumber)).block()!!.first()
+    val currentPrisonName = prisonLookupService.getPrisonName(recall.currentPrison()!!)
+    return DossierContext(recall, prisoner, currentPrisonName)
+  }
+}
+
+data class DossierContext(
+  val recall: Recall,
+  val prisoner: Prisoner,
+  val currentPrisonName: PrisonName,
+) {
+  fun getReasonsForRecallContext(): ReasonsForRecallContext {
+    return ReasonsForRecallContext(
+      FirstAndMiddleNames(FirstName(prisoner.firstName!!), prisoner.middleNames?.let { MiddleNames(it) }),
```

Comment:
feels like we should add a function for prisoner -> FullName and/or FirstAndMiddleNames

Reply:
yes there is one, looks like we didn't use it ;-)